dbms/src/Columns/ColumnString.h

```diff
@@ -120,11 +120,16 @@ class ColumnString final : public IColumn
         }
         else
         {
-            const size_t old_size = chars.size();
-            const size_t size_to_append = src.offsets[0];
-            const size_t new_size = old_size + size_to_append;
-
-            chars.resize(new_size);
+            size_t size_to_append = 0;
+            size_t old_size = chars.size();
+            size_t new_size = old_size;
+            if (src.offsets.size() > 0)
```

Comment:
Why do we need this? If src column is empty it is incorrect to insert n'th row from it. Probably, better to check it before insertion.

Reply:
Already answered in previous request
Without check it gets SEGV

Reply:
Why?

Reply:
Breakpoint 1, DB::ColumnString::insertFrom (this=0xbec38a0, src_=..., n=0) at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/Columns/ColumnString.h:123
warning: Source file is more recent than executable.
123                 size_t size_to_append = 0;
(gdb) p src.offsets
$1 = {<boost::noncopyable_::noncopyable> = {<No data fields>}, <Allocator<false>> = {static clear_memory = false}, static pad_right = <optimized out>,
  c_start = 0x0, c_end = 0x0, c_end_of_storage = 0x0}
(gdb) bt
#0  DB::ColumnString::insertFrom (this=0xbec38a0, src_=..., n=0) at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/Columns/ColumnString.h:123
#1  0x0000000007341459 in DB::MySQLBlockInputStream::insertDefaultValue (column=0xbec38a0, sample_column=...)
    at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/Dictionaries/MySQLBlockInputStream.h:31
#2  0x000000000734021e in DB::MySQLBlockInputStream::readImpl (this=0x7fffeb1cbaa0)
    at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/Dictionaries/MySQLBlockInputStream.cpp:90
#3  0x000000000677165d in DB::IProfilingBlockInputStream::read (this=0x7fffeb1cbaa0)
    at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/DataStreams/IProfilingBlockInputStream.cpp:42
#4  0x00000000066edb47 in DB::TableFunctionMySQL::execute (this=0xc6bda30, ast_function=..., context=...)
    at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/TableFunctions/TableFunctionMySQL.cpp:96
#5  0x0000000007776984 in DB::InterpreterSelectQuery::basicInit (this=0xbeceb00, input=...)
    at /home/aavdonkin/musor/br2/ClickHouse/dbms/src/Interpreters/InterpreterSelectQuery.cpp:137